vendor/k8s.io/kubernetes/pkg/kubectl/cmd/annotate.go

```diff
@@ -205,6 +200,22 @@ func (o AnnotateOptions) RunAnnotate() error {
 		return err
 	}
 
+	rInfos, err := r.Infos()
+	if err != nil {
+		return err
+	}
+
+	// number of resources found
+	numResources := len(rInfos)
```

Comment:
Getting 1 item back from `oc get pods` should not trigger this behavior.  resource.Builder tracks whether the user asks for `oc get pods bar` via the Singular flag.